catalog/to-consul/syncer.go

```diff
@@ -169,31 +194,31 @@ func (s *ConsulSyncer) watchReapableServices(ctx context.Context) {
 			return
 		}
 
-		// Update our blocking index
-		opts.WaitIndex = meta.LastIndex
-
-		// Lock so we can modify the
+		// Lock so we can modify the stored state
 		s.lock.Lock()
 
 		// Go through the service map and find services that should be reaped
-		for name, tags := range serviceMap {
-			for _, tag := range tags {
-				if tag == s.ConsulK8STag {
-					// We only care if we don't know about this service at all.
-					if _, ok := s.services[name]; ok {
-						continue
+		for ns, services := range serviceMap {
+			for name, tags := range services {
+				for _, tag := range tags {
+					if tag == s.ConsulK8STag {
+						// We only care if we don't know about this service at all.
+						if s.serviceNames[ns].Contains(name) {
+							continue
+						}
+
+						s.Log.Info("invalid service found, scheduling for delete",
+							"service-name", name, "service-consul-namespace", ns)
```

Comment:
```suggestion
							"name", name, "consul-ns", ns)
```

That the name is of the service is implied since it says "invalid service found". Probably best to keep these short if possible so logs are more concise